dev/com.ibm.ws.install/src/com/ibm/ws/install/internal/VerifySignatureUtility.java

```diff
@@ -0,0 +1,364 @@
+
+import org.bouncycastle.jce.provider.BouncyCastleProvider;
+import org.bouncycastle.openpgp.PGPCompressedData;
+import org.bouncycastle.openpgp.PGPException;
+import org.bouncycastle.openpgp.PGPPublicKey;
+import org.bouncycastle.openpgp.PGPPublicKeyRing;
+import org.bouncycastle.openpgp.PGPPublicKeyRingCollection;
+import org.bouncycastle.openpgp.PGPSignature;
+import org.bouncycastle.openpgp.PGPSignatureList;
+import org.bouncycastle.openpgp.PGPUtil;
+import org.bouncycastle.openpgp.jcajce.JcaPGPObjectFactory;
+import org.bouncycastle.openpgp.operator.jcajce.JcaKeyFingerprintCalculator;
+import org.bouncycastle.openpgp.operator.jcajce.JcaPGPContentVerifierBuilderProvider;
+
+import com.ibm.ws.install.InstallConstants;
+import com.ibm.ws.install.InstallConstants.VerifyOption;
+import com.ibm.ws.install.InstallException;
+import com.ibm.ws.install.internal.InstallLogUtils.Messages;
+import com.ibm.ws.kernel.boot.cmdline.Utils;
+
+/**
+ *
+ */
+public class VerifySignatureUtility {
+
+    private static final Logger logger = InstallLogUtils.getInstallLogger();
+    private static final String DEFAULT_LIBERTY_KEY_ID = "0x4D210F6946102B8E";
+    private static final String UBUNTU_HOST = "keyserver.ubuntu.com";
+    private static final String PGP_HOST = "keys.openpgp.org";
+    private static final String MIT_HOST = "pgp.mit.edu";
+    private static final String UbuntuServerURL = "https://keyserver.ubuntu.com/pks/lookup?op=get&options=mr&search=";
+    private static final String MITServerURL = "https://pgp.mit.edu/pks/lookup?op=get&options=mr&search=";
+    private static final String PGPServerURL = "https://keys.openpgp.org/pks/lookup?op=get&options=mr&search=";
+
+    private final ProgressBar progressBar = ProgressBar.getInstance();
+
+    VerifySignatureUtility() {
+        Security.addProvider(new BouncyCastleProvider());
+    }
+
+    public boolean isLibertyKeyValid() {
+        try {
+            //TODO: change to valid file path
+            File LibertyKey = new File("/some_path_to_liberty_key");
+            InputStream keyIn = new BufferedInputStream(new FileInputStream(LibertyKey));
+            PGPPublicKeyRing pgpPubKeyRing = new PGPPublicKeyRing(PGPUtil.getDecoderStream(keyIn), new JcaKeyFingerprintCalculator());
+            if (pgpPubKeyRing.getPublicKey().hasRevocation()) {
+                logger.warning(Messages.INSTALL_KERNEL_MESSAGES.getLogMessage("ERROR_REVOKED_PUBLIC_KEY", String.format("%x", pgpPubKeyRing.getPublicKey().getKeyID())));
+                return false;
+            }
+        } catch (IOException e) {
+            logger.fine("Liberty key is corrupted or revoked. Downloading from the keyserver..");
+            return false;
+        }
+
+        return true;
+    }
+
+    public List<File> downloadPublicKeys(Collection<Map<String, String>> keys, VerifyOption verify, Map<String, Object> envMap) throws InstallException {
+        List<String> allKeys = getPublicKeyURL(keys, verify);
```

Comment:
let's return a List<URL> here instead since we iterate through this list and create new URL objects on line 101 anyways